### PR TITLE
[video_player] Apply common texture API

### PR DESCRIPTION
**Pull request by xiaowei-guan**

### CODE REVIEW

**Comment by swift-kim** [comment]
Please update `pubspec.yaml` (package version), `README.md` (package version in Usage), and `CHANGELOG.md` properly if you want this change to be published on pub.dev.

**Comment by swift-kim** [comment]
Frame rendering is a little bit unstable (unordered) on one of my watch devices.

[video_player_tizen.zip](https://github.com/flutter-tizen/plugins/files/6582239/video_player_tizen.zip)

```
NAME=Tizen
VERSION="6.0.0 (Tizen6/Unified)"
ID=tizen
VERSION_ID=6.0.0
PRETTY_NAME="Tizen 6.0.0 (Tizen6/Unified)"
ANSI_COLOR="0;36"
CPE_NAME="cpe:/o:tizen:tizen:6.0.0"
BUILD_ID=tizen-6.0-unified_20201216.3_wearable-wayland-armv7l-tw3
```

It's not reproducible on other devices though. So never mind.

File: packages/video_player/tizen/src/video_player.h [hunk 1]
```diff
@@ -48,7 +50,11 @@ class VideoPlayer {
   std::unique_ptr<flutter::EventChannel<flutter::EncodableValue>> eventChannel_;
   std::unique_ptr<flutter::EventSink<flutter::EncodableValue>> eventSink_;
   long textureId_;
-  FlutterTextureRegistrar *textureRegistrar_;
+  flutter::TextureRegistrar *textureRegistrar_;
+  std::unique_ptr<flutter::TextureVariant> textureVariant_;
+  std::unique_ptr<FlutterDesktopGpuBuffer> flutterDesktopGpuBuffer_;
+  std::mutex mutex_;
+  media_packet_h mediaPacket_ = nullptr;
```
Comment (bbrto21) [inline packages/video_player/tizen/src/video_player.h:57]:
```suggestion
  flutter::TextureRegistrar *texture_registrar_;
  std::unique_ptr<flutter::TextureVariant> texture_variant_;
  std::unique_ptr<FlutterDesktopGpuBuffer> flutter_desktop_gpu_buffer_;
  std::mutex mutex_;
  media_packet_h media_packet_ = nullptr;
```

**Review by bbrto21 — COMMENTED** [review]
Other things looks good to me

File: packages/video_player/tizen/src/video_player.h (hunk 1 above)
Comment (xiaowei-guan) [reply packages/video_player/tizen/src/video_player.h]: Thanks for your code review, I have update variable name.

File: packages/video_player/tizen/src/video_player.cc [hunk 2]
```diff
@@ -49,14 +49,53 @@ static std::string StateToString(player_state_e state) {
...
+  std::lock_guard<std::mutex> lock(mutex_);
+  if (media_packet_) {
+    media_packet_destroy(media_packet_);
+    media_packet_ = nullptr;
+  }
+}
+
 VideoPlayer::VideoPlayer(flutter::PluginRegistrar *pluginRegistrar,
-                         FlutterTextureRegistrar *textureRegistrar,
+                         flutter::TextureRegistrar *textureRegistrar,
```
Comment (bbrto21) [inline packages/video_player/tizen/src/video_player.cc:83]:
```suggestion
                         flutter::TextureRegistrar *texture_registrar,
```

File: packages/video_player/tizen/src/video_player.cc [hunk 3]
```diff
@@ -455,14 +495,15 @@ void VideoPlayer::onErrorOccurred(int code, void *data) {
...
     LOG_INFO("[VideoPlayer.onErrorOccurred] send error event");
-    player->eventSink_->Error("Video player had error",
-                              get_error_message(code));
+    player->event_sink_->Error("Video player had error",
+                               get_error_message(code));
   }
 }
 
 void VideoPlayer::onVideoFrameDecoded(media_packet_h packet, void *data) {
+  LOG_DEBUG("onVideoFrameDecoded");
```
Comment (bbrto21) [inline packages/video_player/tizen/src/video_player.cc:506]: IMO, This generates too many logs. Could you remove this line?

**Review by bbrto21 — COMMENTED** [review]
I think that 508-517 lines are no longer needed in video_player.cc

File: packages/video_player/tizen/src/video_player.cc (hunk 3 above)
Comment (xiaowei-guan) [reply packages/video_player/tizen/src/video_player.cc]: ok, I delete it.

File: packages/video_player/tizen/src/video_player.cc (hunk 2 above)
Comment (xiaowei-guan) [reply packages/video_player/tizen/src/video_player.cc]: ok,

**Review by bbrto21 — APPROVED** [review]
LGTM

**Pull request merged**